yt_dlp/extractor/theplatform.py

```diff
@@ -103,6 +105,36 @@ def _add_chapter(start_time, end_time):
             for chapter in tp_chapters[:-1]:
                 _add_chapter(chapter.get('startTime'), chapter.get('endTime'))
             _add_chapter(tp_chapters[-1].get('startTime'), tp_chapters[-1].get('endTime') or duration)
+        info_keywords_str = info.get('keywords', {str_or_none})
+        tags = []
+        if (info_keywords_str is not None) and (info_keywords_str != ''):
+            tags = re.split(', |; |,', info_keywords_str)
+        location = None
+        series = None
+        season_number = None
+        # The following can be uncommented as soon as #7838 is merged:
+        # media_type = None
+        categories = []
+        categories_data = info.get('categories') or []
+        for x in categories_data:
+            if x.get('name') is not None:
+                # Sometimes, there will be several kinds of categories
+                # in this case, it will have a label field with a value 'category'
+                if (x.get('label') is None) or (x.get('label') == 'category'):
+                    categories.append(x.get('name'))
```

Comment:
we can use traversal filtering to simplify this part
```suggestion
        categories = traverse_obj(info, (
            'categories', lambda _, v: v.get('label') in ('category', None), 'name', {str})) or None
```
we can inline this in the return too